internal/service/advancedclustertpf/model_ClusterDescriptionProcessArgs20240805.go

```diff
@@ -44,9 +46,25 @@ func AddAdvancedConfig(ctx context.Context, tfModel *TFModel, input *admin.Clust
 			SampleSizeBiconnector:            types.Int64Value(conversion.SafeValue(conversion.IntPtrToInt64Ptr(input.SampleSizeBIConnector))),
 			SampleRefreshIntervalBiconnector: types.Int64Value(conversion.SafeValue(conversion.IntPtrToInt64Ptr(input.SampleRefreshIntervalBIConnector))),
 			TransactionLifetimeLimitSeconds:  types.Int64Value(conversion.SafeValue(input.TransactionLifetimeLimitSeconds)),
+			DefaultMaxTimeMS:                 types.Int64PointerValue(conversion.IntPtrToInt64Ptr(input.DefaultMaxTimeMS)),
+			TlsCipherConfigMode:              types.StringValue(conversion.SafeValue(input.TlsCipherConfigMode)),
+			CustomOpensslCipherConfigTls12:   customOpensslCipherConfigTLS12(ctx, input, diags),
 		}
+	} else {
+		advancedConfig.CustomOpensslCipherConfigTls12 = types.SetNull(types.StringType) // required to handle move state
 	}
+
 	objType, diagsLocal := types.ObjectValueFrom(ctx, AdvancedConfigurationObjType.AttrTypes, advancedConfig)
 	diags.Append(diagsLocal...)
 	tfModel.AdvancedConfiguration = objType
 }
+
+func customOpensslCipherConfigTLS12(ctx context.Context, processArgs *admin.ClusterDescriptionProcessArgs20240805, diags *diag.Diagnostics) types.Set {
+	if processArgs != nil && len(*processArgs.CustomOpensslCipherConfigTls12) == 0 {
```

Comment:
```suggestion
	if processArgs == nil || len(*processArgs.CustomOpensslCipherConfigTls12) == 0 {
```

I think we want this condition here, or not sure what we want to do when processArgs is/is not nil

Reply:
you're right, updated the condition

Reply:
can we have a test for that passes now, but wouldn't have passed before this change?   to make sure the implementation logic is correct now, and we don't change it later by mistake while doing some code changes

Reply:
Reg tests: in general I went to check and it looks like we only added one test check in this PR, do we need more tests or are we actually leveraging the already existing ones you created before?

Reply:
@lantoli Due to this above condition you provided I was able to refactor the original logic a bit. Previously I had to add an additional `else` block such as [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2972/commits/76212a3f7abd0351b30490c3831c3b319d209809#diff-bef1833adcf6508cb5d2efe463fffb92a315ded67bf2743550d5f1bfe178ab9c). Now due to the updated condition we don't need that anymore. If the condition is changed for some reason the move block tests will fail so we should have that covered.

@marcosuma As for tests in general, I think the current tests `TestAccClusterAdvancedCluster_advancedConfig_oldMongoDBVersion`, `TestAccClusterAdvancedCluster_advancedConfig` and `TestAccClusterAdvancedCluster_defaultWrite` provide quite decent coverage for the attributes added in this PR already for null, default or setting/updating values. Please let me know if there's any additional test case I can handle to improve.